cla_backend/libs/eligibility_calculator/calculator.py

```diff
@@ -87,14 +86,14 @@ def _do_cfe_civil_check(self):
 
     @staticmethod
     def _is_data_complete_enough_to_call_cfe(case_data):
-        if not hasattr(case_data, "facts"):
+        if not "facts" in case_data.__dict__:
```

Comment:
Looks good! However it's a bit surprising, so how about preceding this with a comment explaining it (and a docstring why not...):
```
# The CFE means calculation doesn't always need all the case_data fields known. Here we check for the presence of certain fields.
# We can't use hasattr(case_data, "<field_name>") to know if a field has been set, because hasattr() under the hood calls getattr(), and we've overridden ModelMixin.__getattr__() to return a default value when the field is absent. So instead, we check for the key in the case_data.__dict__
```